pkg/controllers/provisioningv2/rke2/provisioningcluster/controller.go

```diff
@@ -181,6 +181,13 @@ func (h *handler) OnChange(_ string, cluster *rancherv1.Cluster) (*rancherv1.Clu
 		if machinePool.DynamicSchemaSpec != "" && json.Unmarshal([]byte(machinePool.DynamicSchemaSpec), &spec) == nil {
 			continue
 		}
+		apiVersion := machinePool.NodeConfig.APIVersion
+		if apiVersion == "" {
+			apiVersion = rke2.DefaultMachineConfigAPIVersion
+		}
+		if machinePool.NodeConfig != nil && apiVersion != rke2.DefaultMachineConfigAPIVersion {
+			continue
+		}
```

Comment:
1. If you need to check if `machinePool.NodeConfig` is nil, should you do it before getting "machinePool.NodeConfig.APIVersion"? 
2.  Can you confirm that it is expected to not set `APIVersion` back to the object when it is empty?

Reply:
re: 1 -- good catch
re: 2 -- this controller should not be manipulating the source object (the `clusters.provisioning.cattle.io/v1` object), so I would not expect it to set the api version on it.